### ISSUE TITLE
Add graphql 16.6.0 to peerDependencies in @accounts/graphql-api

### ISSUE THREAD

**Issue by danyhiol** [accounts-js/accounts#1244]
# Bug report

## Describe the bug

Appolo recently released `@apollo/server v4`. Doing this, they dropped support for `Graphql < 16.6.0`.
`accounts-graphql-api` uses as peerDependencies `"graphql": "^14.6.0 || ^15.0.0",` this causes all kinds of problems working with the new apollo-server version.

## To Reproduce

Try installing @apollo/server@4.0.0 when using `@accounts/graphql-api`

## Expected behavior

Avoid dependencies conflict by updating graphql.

## Screenshots
`> npm ls graphql`
 ```js
@accounts/graphql-api@0.33.2
│ ├─┬ @graphql-tools/merge@6.2.13
│ │ ├─┬ @graphql-tools/schema@7.1.5
│ │ │ └── graphql@15.8.0 deduped invalid: "^16.6.0" from node_modules/@apollo/server, "^16.6.0" from node_modules/type-graphql
│ │ └── graphql@15.8.0 deduped invalid: "^16.6.0" from node_modules/@apollo/server, "^16.6.0" from node_modules/type-graphql
│ ├─┬ @graphql-tools/utils@7.9.0
│ │ └── graphql@15.8.0 deduped invalid: "^16.6.0" from node_modules/@apollo/server, "^16.6.0" from node_modules/type-graphql
│ └── graphql@15.8.0 deduped
├─┬ @accounts/graphql-client@0.33.1
│ ├─┬ @graphql-typed-document-node/core@3.1.0
│ │ └── graphql@15.8.0 deduped invalid: "^16.6.0" from node_modules/@apollo/server, "^16.6.0" from node_modules/type-graphql
│ └── graphql@15.8.0 deduped
├─┬ @apollo/client@3.6.9
│ ├─┬ @graphql-typed-document-node/core@3.1.1
│ │ └── graphql@15.8.0 deduped invalid: "^16.6.0" from node_modules/@apollo/server, "^16.6.0" from node_modules/type-graphql
│ └── graphql@15.8.0 deduped
├─┬ @apollo/server@4.2.1
│ ├─┬ @apollo/cache-control-types@1.0.2
│ │ └── graphql@15.8.0 deduped invalid: "^16.6.0" from node_modules/@apollo/server, "^16.6.0" from node_modules/type-graphql
│ ├─┬ @apollo/server-gateway-interface@1.0.7
│ │ └── graphql@15.8.0 deduped invalid: "^16.6.0" from node_modules/@apollo/server, "^16.6.0" from node_modules/type-graphql
│ ├─┬ @apollo/utils.usagereporting@2.0.0
│ │ ├─┬ @apollo/utils.dropunuseddefinitions@2.0.0
│ │ │ └── graphql@15.8.0 deduped invalid: "^16.6.0" from node_modules/@apollo/server, "^16.6.0" from node_modules/type-graphql
│ │ ├─┬ @apollo/utils.printwithreducedwhitespace@2.0.0
│ │ │ └── graphql@15.8.0 deduped invalid: "^16.6.0" from node_modules/@apollo/server, "^16.6.0" from node_modules/type-graphql
│ │ ├─┬ @apollo/utils.removealiases@2.0.0
│ │ │ └── graphql@15.8.0 deduped invalid: "^16.6.0" from node_modules/@apollo/server, "^16.6.0" from node_modules/type-graphql
│ │ ├─┬ @apollo/utils.sortast@2.0.0
│ │ │ └── graphql@15.8.0 deduped invalid: "^16.6.0" from node_modules/@apollo/server, "^16.6.0" from node_modules/type-graphql
│ │ ├─┬ @apollo/utils.stripsensitiveliterals@2.0.0
│ │ │ └── graphql@15.8.0 deduped invalid: "^16.6.0" from node_modules/@apollo/server, "^16.6.0" from node_modules/type-graphql
│ │ └── graphql@15.8.0 deduped invalid: "^16.6.0" from node_modules/@apollo/server, "^16.6.0" from node_modules/type-graphql
│ └── graphql@15.8.0 deduped invalid: "^16.6.0" from node_modules/@apollo/server
├─┬ @graphql-modules/core@0.7.17
│ ├─┬ @graphql-toolkit/common@0.10.6
│ │ └── graphql@15.8.0 deduped invalid: "^16.6.0" from node_modules/@apollo/server, "^16.6.0" from node_modules/type-graphql
│ ├─┬ @graphql-toolkit/schema-merging@0.10.6
│ │ └── graphql@15.8.0 deduped invalid: "^16.6.0" from node_modules/@apollo/server, "^16.6.0" from node_modules/type-graphql
│ ├─┬ graphql-tools@5.0.0
│ │ ├─┬ apollo-link@1.2.14
│ │ │ ├─┬ apollo-utilities@1.3.4
│ │ │ │ └── graphql@15.8.0 deduped invalid: "^16.6.0" from node_modules/@apollo/server, "^16.6.0" from node_modules/type-graphql
│ │ │ └── graphql@15.8.0 deduped invalid: "^16.6.0" from node_modules/@apollo/server, "^16.6.0" from node_modules/type-graphql
│ │ ├─┬ apollo-upload-client@13.0.0
│ │ │ ├─┬ apollo-link-http-common@0.2.16
│ │ │ │ └── graphql@15.8.0 deduped invalid: "^16.6.0" from node_modules/@apollo/server, "^16.6.0" from node_modules/type-graphql
│ │ │ └─┬ apollo-link@1.2.14
│ │ │   ├─┬ apollo-utilities@1.3.4
│ │ │   │ └── graphql@15.8.0 deduped invalid: "^16.6.0" from node_modules/@apollo/server, "^16.6.0" from node_modules/type-graphql
│ │ │   └── graphql@15.8.0 deduped invalid: "^16.6.0" from node_modules/@apollo/server, "^16.6.0" from node_modules/type-graphql
│ │ └── graphql@15.8.0 deduped invalid: "^16.6.0" from node_modules/@apollo/server, "^16.6.0" from node_modules/type-graphql
│ └── graphql@15.8.0 deduped invalid: "^16.6.0" from node_modules/@apollo/server
├─┬ @graphql-tools/merge@8.3.12
│ ├─┬ @graphql-tools/utils@9.1.1
│ │ └── graphql@15.8.0 deduped invalid: "^16.6.0" from node_modules/@apollo/server, "^16.6.0" from node_modules/type-graphql
│ └── graphql@15.8.0 deduped invalid: "^16.6.0" from node_modules/@apollo/server
├─┬ @graphql-tools/schema@9.0.10
│ ├─┬ @graphql-tools/utils@9.1.1
│ │ └── graphql@15.8.0 deduped invalid: "^16.6.0" from node_modules/@apollo/server, "^16.6.0" from node_modules/type-graphql
│ └── graphql@15.8.0 deduped invalid: "^16.6.0" from node_modules/@apollo/server
├─┬ @graphql-tools/utils@8.13.1
│ └── graphql@15.8.0 deduped invalid: "^16.6.0" from node_modules/@apollo/server
├─┬ apollo-link-http@1.5.17
│ ├─┬ apollo-link-http-common@0.2.16
│ │ └── graphql@15.8.0 deduped invalid: "^16.6.0" from node_modules/@apollo/server, "^16.6.0" from node_modules/type-graphql
│ ├─┬ apollo-link@1.2.14
│ │ ├─┬ apollo-utilities@1.3.4
│ │ │ └── graphql@15.8.0 deduped invalid: "^16.6.0" from node_modules/@apollo/server, "^16.6.0" from node_modules/type-graphql
│ │ └── graphql@15.8.0 deduped invalid: "^16.6.0" from node_modules/@apollo/server, "^16.6.0" from node_modules/type-graphql
│ └── graphql@15.8.0 deduped invalid: "^16.6.0" from node_modules/@apollo/server
├─┬ graphql-scalars@1.20.1
│ └── graphql@15.8.0 deduped invalid: "^16.6.0" from node_modules/@apollo/server
├─┬ graphql-subscriptions@2.0.0
│ └── graphql@15.8.0 deduped invalid: "^16.6.0" from node_modules/@apollo/server
├─┬ graphql-tag@2.12.6
│ └── graphql@15.8.0 deduped invalid: "^16.6.0" from node_modules/@apollo/server
├─┬ graphql-ws@5.11.2
│ └── graphql@15.8.0 deduped invalid: "^16.6.0" from node_modules/@apollo/server
├── graphql@15.8.0 invalid: "^16.6.0" from node_modules/@apollo/server
└─┬ type-graphql@2.0.0-beta.1
  ├─┬ graphql-query-complexity@0.12.0
  │ └── graphql@15.8.0 deduped invalid: "^16.6.0" from node_modules/@apollo/server, "^16.6.0" from node_modules/type-graphql
  └── graphql@15.8.0 deduped invalid: "^16.6.0" from node_modules/@apollo/server, "^16.6.0" from node_modules/type-graphql

```
## System information

-` Version of accounts-js: [0.33.2]`  `"@accounts/graphql-api": "^0.33.2"`

**Comment by KeithGillette** [comment]
I believe supporting `graphql@^16.0.0` will also required updating types, per #1226.

**Comment by dglyan** [comment]
Hi! Were there any solutions to this problem?

**Comment by pradel** [comment]
@dglyan this should be fixed by the 1.0 release https://github.com/accounts-js/accounts/pull/1258.
You can maybe try the beta release for now, would that work @darkbasic ?

**Comment by darkbasic** [comment]
Yep we use latest graphql everywhere in 1.0!
